File: smartcity_highfive/HighFive.SmartCity.Domain.Entity/UsuarioOferta.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace HighFive.SmartCity.Domain.Entity
{
    [Table("tbl_usuario_oferta")]
    public class UsuarioOferta
    {
		public UsuarioOferta()
        {
        }

		[Column("id_oferta")]
		[Key]
		[Required]
		[DatabaseGenerated(DatabaseGeneratedOption.Identity)]
		public int Id { get; set; }

		//@ManyToOne
		[Column("id_usuario")]
		[Required]
		public Usuario Usuario { get; set; }

		//@ManyToOne
		[Column("id_categoriaoferta")]
		[Required]
		public CategoriaOferta Categoria { get; set; }

		//@OneToMany(mappedBy = "oferta")
		public ICollection<UsuarioOfertaDisponibilidade> DatasDisponiveis { get; set; }

		[Column("nm_oferta")]
		[MaxLength(250)]
		[Required]
		public string Oferta { get; set; }

		[Column("ds_oferta")]
		[MaxLength(800)]
		public string Descricao { get; set; }


		[Column("qt_horas_duracao")]
		[Required]
		public int HorasDuracao { get; set; }

		[Column("vl_pontos")]
		[Required]
		public decimal Pontos { get; set; }

		[Column("dh_criacao")]
		[Required]
		public DateTime DataCriacao { get; set; }

		[Column("dh_atualizacao")]
		[Required]
		public DateTime DataAtualizacao { get; set; }
	}
}

File: SmartCityApp/app/src/main/java/com/br/smartcityapp/entity/UsuarioOferta.java
package com.br.smartcityapp.entity;

import java.util.Date;

public class UsuarioOferta {

    private int id;
    //private Usuario Usuario;
    //private CategoriaOferta Categoria;
    //private ICollection<UsuarioOfertaDisponibilidade> DatasDisponiveis;
    private String oferta;
    private String descricao;
    private int horasDuracao;
    private float pontos;
    private Date dataCriacao;
    private Date dataAtualizacao;


    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public String getOferta() {
        return oferta;
    }

    public void setOferta(String oferta) {
        this.oferta = oferta;
    }

    public String getDescricao() {
        return descricao;
    }

    public void setDescricao(String descricao) {
        this.descricao = descricao;
    }

    public int getHorasDuracao() {
        return horasDuracao;
    }

    public void setHorasDuracao(int horasDuracao) {
        this.horasDuracao = horasDuracao;
    }

    public float getPontos() {
        return pontos;
    }

    public void setPontos(float pontos) {
        this.pontos = pontos;
    }

    public Date getDataCriacao() {
        return dataCriacao;
    }

    public void setDataCriacao(Date dataCriacao) {
        this.dataCriacao = dataCriacao;
    }

    public Date getDataAtualizacao() {
        return dataAtualizacao;
    }

    public void setDataAtualizacao(Date dataAtualizacao) {
        this.dataAtualizacao = dataAtualizacao;
    }
}
